lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -745,6 +774,73 @@ public function computeChangeSets()
         }
     }
 
+    private function computeToManyAssociationChanges(ToManyAssociationMetadata $association, $value)
```

Comment:
This code is mostly duplicated/extracted from `computeAssociationChanges`. The code duplication can be massively reduced if we use dedicated iterators over property types.